File: OrienGO-Frontend/src/app/model/dto/CoachReturn.dto.ts
import { AccountPrivacy } from "../enum/AccountPrivacy.enum";
import { CoachSpecialization } from "../enum/CoachSpecialization.enum";
import { GenderType } from "../enum/GenderType.enum";
import { MessagePermission } from "../enum/MessagePermission.enum";
import { VisibilityStatus } from "../enum/VisibilityStatus.enum";
import { LocationDTO } from "./Location.dto";

export interface CoachReturnDTO {
  // ==============================
  // User (parent class) fields
  // ==============================
  id: number;

  firstName: string;
  lastName: string;
  age: number;
  gender: GenderType;        // enum
  phoneNumber: string;
  email: string;
  enabled: boolean;
  suspended: boolean;
  suspensionReason: string;
  suspendedUntil: string;    // ISO string (Java LocalDateTime → string)
  tokenExpired: boolean;

  // ==============================
  // Coach-specific fields
  // ==============================
  profileVisibility: VisibilityStatus; // enum
  location: LocationDTO;               // another DTO
  messagePermission: MessagePermission; // enum
  accountPrivacy: AccountPrivacy;      // enum
  specialization: CoachSpecialization; // enum
  rate: number;
  expertise: string;
  services: string;
  availability: string;

  // ==============================
  // Audit fields
  // ==============================
  createdAt: string;   // Java LocalDateTime → string
  updatedAt: string;   // Java LocalDateTime → string
}

File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/CoachReturnDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Set;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CoachReturnDTO {

    // ==============================
    // User (parent class) fields
    // ==============================
    private Long id;

    private String firstName;

    private String lastName;

    private Integer age;

    private GenderType gender;

    private String phoneNumber;

    private String email;

    private boolean enabled;

    private boolean suspended;

    private String suspensionReason;

    private LocalDateTime suspendedUntil;

    private Boolean tokenExpired;

    // ==============================
    // Coach-specific fields
    // ==============================
    private VisibilityStatus profileVisibility;

    private LocationDTO location;

    private MessagePermission messagePermission;

    private AccountPrivacy accountPrivacy;

    private CoachSpecialization specialization;

    private Double rate;

    private String expertise;

    private String services;

    private String availability;

    // ==============================
    // Audit fields
    // ==============================
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

}
